### ISSUE TITLE
bug: operator < is parsed as begnning of template argument list.

### ISSUE THREAD

**Issue by scturtle** [tree-sitter/tree-sitter-cpp#269]
### Did you check existing issues?

- [ ] I have read all the [tree-sitter docs](https://tree-sitter.github.io/tree-sitter/using-parsers) if it relates to using the parser
- [X] I have searched the existing issues of tree-sitter-cpp

### Tree-Sitter CLI Version, if relevant (output of `tree-sitter --version`)

tree-sitter 0.22.6

### Describe the bug

Operator `<` is parsed as beginning of template argument list.

### Steps To Reproduce/Bad Parse Tree

```
(translation_unit
 (function_definition type: (primitive_type)
  declarator: 
   (function_declarator declarator: (identifier)
    parameters: (parameter_list ( )))
  body: 
   (compound_statement {
    (declaration
     type: 
      (struct_specifier struct name: (type_identifier)
       body: 
        (field_declaration_list {
         (field_declaration type: (primitive_type) declarator: (field_identifier) ;)
         }))
     declarator: (identifier) ;)
    (declaration type: (primitive_type)
     declarator: 
      (init_declarator declarator: (identifier) =
       value: 
        (field_expression argument: (identifier) operator: .
         field: 
          (template_method name: (field_identifier)
           arguments: (template_argument_list < (number_literal) >))))
     ;)
    })))
```

### Expected Behavior/Parse Tree

```
(translation_unit
 (function_definition type: (primitive_type)
  declarator: 
   (function_declarator declarator: (identifier)
    parameters: (parameter_list ( )))
  body: 
   (compound_statement {
    (declaration
     type: 
      (struct_specifier struct name: (type_identifier)
       body: 
        (field_declaration_list {
         (field_declaration type: (primitive_type) declarator: (field_identifier) ;)
         }))
     declarator: (identifier) ;)
    (declaration type: (primitive_type)
     declarator: 
      (init_declarator declarator: (identifier) =
       value: 
        (binary_expression
         left: (field_expression argument: (identifier) operator: . field: (field_identifier))
         operator: < right: (number_literal)))
     ;)
    })))
```

### Repro

```cpp
int main() {
  struct A {int b;} a;
  bool t = a.b < 3;
}
```

**Comment by touzeauv** [comment]
You ran into the same problem that is described in issue https://github.com/tree-sitter/tree-sitter-cpp/issues/231.
I suggested a fixed (see pull request https://github.com/tree-sitter/tree-sitter-cpp/pull/267) that solves your problem (it produces the same tree you expected).

**Comment by jdrouhard** [comment]
Fixed by #267
